zerver/lib/actions.py

```diff
@@ -1284,6 +1285,93 @@ def do_delete_user(user_profile: UserProfile) -> None:
         )
 
 
+def do_delete_user_preserving_messages(user_profile: UserProfile) -> None:
```

Comment:
I chose this name for now, instead of the initially suggested `do_scrub_user`, because we already have `do_scrub_realm` and it's not analogical to this function at all, since it actually is meant to purge as much data as possible.

Reply:
Yeah, I think that's a good name.